test/navigation.spec.js

```diff
@@ -151,16 +151,22 @@ describe('navigation', function () {
 
       // Make sure that network events do not emit 'undefined'.
       // @see https://crbug.com/750469
-      page.on('request', (request) => expect(request).toBeTruthy());
-      page.on('requestfinished', (request) => expect(request).toBeTruthy());
-      page.on('requestfailed', (request) => expect(request).toBeTruthy());
+      const requests = [];
+      page.on('request', (request) => requests.push('request'));
+      page.on('requestfinished', (request) => requests.push('requestfinished'));
+      page.on('requestfailed', (request) => requests.push('requestfailed'));
```

Comment:
@whimboo to check my understanding; do you think this test used to pass because we've never get a `requestfinished` event and therefore that expectation would never run?

@mathiasbynens do you have any thoughts on if we should be getting the `requestfinished` event? I think it makes sense that we don't, given that this test checks that requests fail and therefore I wouldn't expect the `requestfinished` event necessarily. I think this was a mistake in the test that made it in and happened to not cause any issues.

Reply:
Yes, exactly. I think we either get `requestfinished` or `requestfailed`. But agreed that this is something @mathiasbynens should reply on.